link_linux.go

```diff
@@ -455,23 +461,43 @@ func LinkAdd(link Link) error {
 		if tuntap.Mode < syscall.IFF_TUN || tuntap.Mode > syscall.IFF_TAP {
 			return fmt.Errorf("Tuntap.Mode %v unknown!", tuntap.Mode)
 		}
-		file, err := os.OpenFile("/dev/net/tun", os.O_RDWR, 0)
-		if err != nil {
-			return err
+
+		queues := tuntap.Queues
+		if queues == 0 { //Legacy compatibility
+			queues = 1
 		}
-		defer file.Close()
-		var req ifReq
-		req.Flags |= syscall.IFF_ONE_QUEUE
-		req.Flags |= syscall.IFF_TUN_EXCL
-		copy(req.Name[:15], base.Name)
-		req.Flags |= uint16(tuntap.Mode)
-		_, _, errno := syscall.Syscall(syscall.SYS_IOCTL, file.Fd(), uintptr(syscall.TUNSETIFF), uintptr(unsafe.Pointer(&req)))
-		if errno != 0 {
-			return fmt.Errorf("Tuntap IOCTL TUNSETIFF failed, errno %v", errno)
+
+		tuntap.Fds = nil
+		for i := 0; i < queues; i++ {
+			file, err := os.OpenFile("/dev/net/tun", os.O_RDWR, 0)
+			if err != nil {
+				cleanupFds(tuntap.Fds[:i])
+				return err
+			}
+			tuntap.Fds = append(tuntap.Fds, file)
+			var req ifReq
+			//req.Flags |= syscall.IFF_ONE_QUEUE
+			//req.Flags |= syscall.IFF_TUN_EXCL
```

Comment:
No. I plan to remove them in the final update. The original logic was setting them. This was just for reference. libvirt does not set them either.